drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-backend/src/main/java/org/drools/workbench/screens/scenariosimulation/backend/server/DMNTypeServiceImpl.java

```diff
@@ -275,6 +285,15 @@ protected void checkTypeSupport(DMNType type,
                                     boolean alreadyInCollection,
                                     ErrorHolder errorHolder,
                                     String fullPropertyPath) {
+        internalCheckTypeSupport(type, alreadyInCollection, errorHolder, fullPropertyPath, new HashSet<>());
+    }
+
+    protected void internalCheckTypeSupport(DMNType type,
+                                            boolean alreadyInCollection,
+                                            ErrorHolder errorHolder,
+                                            String fullPropertyPath,
+                                            Set<String> alreadyVisited) {
+        alreadyVisited.add(type.getName());
```

Comment:
@danielezonca 
if this "alreadyVisited" is the same as "visited" elsewhere, I think it would be better to to call it always the same way (again, I do not remember who originally wrote that).

Reply:
Correct, renamed to `alreadyVisited` :+1: